### PR TITLE
godot4(-mono): Add version rc6

### LINKED ISSUE #994: [Request]: Godot 4.0 (Non-Mono)
### Prerequisites

- [X] I have searched all issues/PRs to ensure it has not already been reported or fixed.

### Criteria

- [X] English interface (or at least English documentation)
- [X] Alternative, forked or prerelease version of an existing package
- [X] Fairly standard install (e.g. no elaborate pre/post install scripts)

### Name

godot4

### Description

A single package to account for all Godot 4.0 non-mono versions (alpha, beta, release candidate).

### Homepage

https://godotengine.org/

### Download Link(s)

https://downloads.tuxfamily.org/godotengine/4.0/

### Corresponding package

https://github.com/ScoopInstaller/Extras/blob/master/bucket/godot.json

### Some Indication of Popularity/Repute

`57.6k Stars` `10k Forks` https://github.com/godotengine/godot

### PR DESCRIPTION
**Pull request by jwmcgettigan**
This PR adds the `godot4` and `godot4-mono` manifests (which I'll call the 'godot4 manifests').
These two manifests effectively supplant the need for the `godot-alpha`, `godot-beta`, and `godot-mono-beta` manifests as all alpha and beta versions can be retrieved with these godot4 manifests.  The only stipulation being that most alpha versions do not include a mono build.

My reason for creating these godot4 manifests was due to the latest Godot 4 versions being release candidates instead of beta builds.  These manifests should theoretically be able to support any new Godot 4 versions even once officially released, although some adjustments to the checkver script may be necessary.

Closes #994 and #995

- [x] I have read the [Contributing Guide](https://github.com/ScoopInstaller/.github/blob/main/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by jwmcgettigan** [comment]
These manifests have worked locally this entire time.  For some reason though, they've been failing the `checkver` check.

These are the errors that I initially encountered.
chrome_2023-02-09_00-42-08

After some modifications, only this error remains.
chrome_2023-02-09_01-40-51

**Comment by jwmcgettigan** [comment]
Running this PowerShell script will give you the list of versions that are supported by the manifests.
```
$WebResponse = Invoke-WebRequest('https://downloads.tuxfamily.org/godotengine/4.0/')
$links = $WebResponse.Links | Select-Object -Skip 1
$versions = $links | foreach {$_.innerText}
$sortedVersions = $versions | Sort-Object {$_ -NotMatch 'pre-'},{"$_" -Replace '\d',''},{("$_" -Replace '\D','') -as [int]}
$sortedVersions
```

Changing the last line to `$sortedVersions | Select-Object -Last 1` will give you the most recent version - just as the manifest does.

**Comment by jwmcgettigan** [comment]
Now no more exceptions, but the version is still returning empty when the script is run through GithubActions even though it works locally.

**godot4-mono**
```
INFO: Checkver

Log of Output:

    GET with 0-byte payload 
    received 6998-byte response of content type text/html 
    godot4-mono:  
    couldn't match '(\S+)' in https://godotengine.org/

INFO: Checkver done
```

**godot4**
```
INFO: Checkver

Log of Output:

    GET with 0-byte payload 
    received 6998-byte response of content type text/html 
    godot4:  
    couldn't match '(\S+)' in https://godotengine.org/

INFO: Checkver done
```

**Comment by deevus** [comment]
Fixed in #1024 

Sorry @jwmcgettigan I went ahead and added Godot RC and didn't notice that you already had an in progress PR for it.

**Comment by jwmcgettigan** [comment]
@deevus I'm fine with closing this if that's what's best but I just wanted to clarify the intent of this PR and its manifests.

The intent is to only need the `godot4` and `godot4-mono` packages instead of constantly needing to swap to another package (e.g. alpha -> beta -> RC -> etc.).  This manifest removes the need for different godot4 packages and allows for alpha, beta, and RC to all be installed from it.  

It also easily accommodates all future versions of Godot 4 as it retrieves the versions directory from the Godot downloads page: https://downloads.tuxfamily.org/godotengine/4.0/

Basically the intent of this PR is to kill two birds with one stone:
- Stop depending on https://godotengine.org/blog/ since it is less consistent and reliable.
- Prevent the need to create new packages/manifests for each version Godot 4.

Perhaps it won't matter anymore after the stable release of Godot 4, but just wanted to be clear as to why I took this route instead of the route you took.

**Comment by deevus** [comment]
You make a good point. Let me look over your changes in greater detail and I'll get back to you.

**Comment by jwmcgettigan** [comment]
@deevus Thanks!  Just to catch you up on the status of the PR, I've had all steps working locally from the very beginning but I can't seem to get the GithubActions bot to pass the `checkver` step.

**Comment by deevus** [comment]
I've updated the manifest to the latest version and updated the branch with master. Let's see if that helps.

**Comment by deevus** [comment]
From https://github.com/ScoopInstaller/Versions/actions/runs/4266748136/jobs/7427627354:

Unsurprisingly it's the same error for both manifests

```
Log of Output:

    GET with 0-byte payload 
    received 7436-byte response of content type text/html 
    godot4-mono:  
    couldn't match '(\S+)' in https://godotengine.org/
```

```
Log of Output:

    GET with 0-byte payload 
    received 7436-byte response of content type text/html 
    godot4:  
    couldn't match '(\S+)' in https://godotengine.org/
```

Looking at the [autoupdate documentation](https://scoop-docs.vercel.app/docs/concepts/App-Manifest-Autoupdate.html#adding-checkver-to-a-manifest), `script` isn't a valid `checkver` option

image

**Comment by deevus** [comment]
It is supported: https://github.com/ScoopInstaller/Scoop/blob/f93028001fbe5c78cc41f59e3814d2ac8e595724/bin/checkver.ps1#L284-L286

**Comment by jwmcgettigan** [comment]
> It is supported: https://github.com/ScoopInstaller/Scoop/blob/f93028001fbe5c78cc41f59e3814d2ac8e595724/bin/checkver.ps1#L284-L286

I was just about to link you to the docs that I've been referencing that do indicate support for `script`:
https://github.com/ScoopInstaller/Scoop/wiki/App-Manifest-Autoupdate

image

**Comment by deevus** [comment]
I'm at a loss as to why this is failing. @rasa?

**Comment by jwmcgettigan** [comment]
Just in case it got buried in the comments, here's the issue I created on the ScoopInstaller/GithubActions repo:
https://github.com/ScoopInstaller/GithubActions/issues/31

I feel like the issue most likely lies there rather than with the script itself.  While it is possible that the script isn't conforming to some unknown requirement, I tried to follow the example of other manifests that use `script` as closely as I can.

**Comment by niheaven** [comment]
Why not check PowerShell... `innerText` is removed in PowerShell and `-replace` needs correct RegEx.

There's nothing to do with GitHubActions.

**Comment by jwmcgettigan** [comment]
@niheaven Thank you for fixing the script!  I wasn't aware that `innerText` was phased out.

**Review by niheaven — APPROVED** [review]

**Comment by deevus** [comment]
@niheaven Autoupdate is broken https://github.com/ScoopInstaller/Versions/commit/91adc5ecb594f74f941188f0ea975de2098d0338

**Comment by niheaven** [comment]
I'll check it

**Comment by niheaven** [comment]
Fix in https://github.com/ScoopInstaller/Versions/commit/eb7f1074486de5d163f9e910b515ba556bd90dc9
